Add notebook example for using Dask EntitySets

thehomebrewnerd opened the pull request:
Adds a new notebook that demonstrates how to compute a feature matrix using Dask EntitySets. 

Fixes #16 

Fixes #19 

Fixes #20

thehomebrewnerd commented:
@rwedge I have removed the original notebook and replaced it with a Dask-only version. Let me know what you think of this setup when you have a chance to review.

rwedge commented:
> @rwedge I have removed the original notebook and replaced it with a Dask-only version. Let me know what you think of this setup when you have a chance to review.

Some sections of the readme that dealt with there being two approaches should be updated

thehomebrewnerd commented:
> > @rwedge I have removed the original notebook and replaced it with a Dask-only version. Let me know what you think of this setup when you have a chance to review.
> 
> Some sections of the readme that dealt with there being two approaches should be updated

Updated to remove reference to multiple notebooks.

rwedge commented on Tutorial.ipynb:
```diff
@@ -213,55 +164,101 @@
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/nate.parsons/dev/featuretools/env/lib/python3.7/site-packages/ipykernel_launcher.py:1: UserWarning: `meta` is not specified, inferred from partial data. Please provide `meta` if the result is unexpected.\n",
```
Is there something we can do to resolve this warning?

thehomebrewnerd replied:
I'll have look into the exact cause of that a little more, but I think it might be something we have fix in Featuretools.

thehomebrewnerd replied:
Disregard my previous comment - I didn't look closely enough and thought that warning was happening somewhere else in the notebook. Yes, I think we can fix this warning.

thehomebrewnerd replied:
@rwedge Added `meta` to the `apply` to fix the previous warning. Notebook should run without any warnings now.

thehomebrewnerd replied:
@rwedge I think everything has been resolved now - are we ready to merge?

thehomebrewnerd commented:
@rwedge I want to take another look at this before we merge it in. In particular, there is this code block at the start of the machine learning step:
```
X = fm_encoded.merge(label_times)
X.drop(["user_id", "time"], axis=1, inplace=True)
X = X.fillna(0)
y = X.pop("bought_product").astype('bool')
```
I'm not sure now why I was merging in the label times df, as the labels are already present in the feature matrix, and then we just drop the user_id and time columns anyway, so that merge isn't doing anything for us.

thehomebrewnerd commented:
> @rwedge I want to take another look at this before we merge it in. In particular, there is this code block at the start of the machine learning step:
> 
> ```
> X = fm_encoded.merge(label_times)
> X.drop(["user_id", "time"], axis=1, inplace=True)
> X = X.fillna(0)
> y = X.pop("bought_product").astype('bool')
> ```
> 
> I'm not sure now why I was merging in the label times df, as the labels are already present in the feature matrix, and then we just drop the user_id and time columns anyway, so that merge isn't doing anything for us.

@rwedge While I was looking into this merge statement, I found another issue that I'm working on resolving. 

First, the merge was not necessary, so that is being removed. The second issue was that when writing the initial feature matrix to disk and then reading it back it, some of the column datatypes were being changed which impacted the encoding and machine learning process. I am updating to skip this intermediate step of writing to disk.

I've made these changes and am now rerunning the notebook to confirm everything works as expected. I hope to have and updated version up for review sometime today.

rwedge commented on README.md:
```diff
@@ -31,18 +31,15 @@ In this tutorial, we show how [Featuretools](https://www.featuretools.com) can b
+
 3. Download the data
 
     You can download the data directly from Instacart [here](https://www.instacart.com/datasets/grocery-shopping-2017).
```
We should update this link as well

thehomebrewnerd replied:
@rwedge Good catch. Updated.

rwedge reviewed (approved):
Looks good!
